[test-triage] OTBN issue with the Python scripts

### Hierarchy of regression failure

Block level

### Failure Description

Similar errors are often raised while running some OTBN tests:
```
* `Job returned non-zero exit code` has 1 failures:
    * Test otbn_zero_state_err_urnd has 1 failures.
        * 0.otbn_zero_state_err_urnd.49783223393755823323792651049911603757994003625678540082947296603666494076901\
         Log /home/dev/src/scratch/master/otbn-sim-xcelium/0.otbn_zero_state_err_urnd/latest/run.log

                    gen_res = self.gen(model, program, end)
                  File "/home/dev/src/hw/ip/otbn/dv/rig/rig/snippet_gens.py", line 167, in gen
                    gen_res = generator.gen(self.gens, model, program)
                  File "/home/dev/src/hw/ip/otbn/dv/rig/rig/gens/branch_gen.py", line 81, in gen
                    off_enc = self.imm_op_type.op_val_to_enc_val(tgt_addr, model.pc)
                  File "/home/dev/src/hw/ip/otbn/util/shared/operand.py", line 430, in op_val_to_enc_val
                    raise ValueError('Cannot encode the value {}{} as a {}-bit '
                ValueError: Cannot encode the value -6410, which encodes to -3205, as a 12-bit signed value. Possible range: -4096..4094.
                ninja: build stopped: subcommand failed.
                make: *** [/home/dev/src/hw/dv/tools/dvsim/sim.mk:52: pre_run] Error 1

```


### Steps to Reproduce

- GitHub Revision: a182fcef27866fb502d8cceb810bdf76cfa3ef42
- dvsim invocation command to reproduce the failure, inclusive of build and run seeds:
util/dvsim/dvsim.py hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson -i otbn_zero_state_err_urnd -t xcelium --fixed-seed 49783223393755823323792651049911603757994003625678540082947296603666494076901

### Tests with similar or related failures

_No response_

--- a/hw/ip/otbn/dv/rig/rig/gens/bad_branch.py
+++ b/hw/ip/otbn/dv/rig/rig/gens/bad_branch.py
@@ -26,6 +26,17 @@ class BadBranch(BranchGen):
                     max_addr: int,
                     model: Model,
                     program: Program) -> int:
+
+        # We only want to pick values that we can actually encode as a 12-bit
+        # signed differences from the current PC (shifted left by 1).
+        min_addr = max(min_addr, model.pc - 4096)
+        max_addr = min(max_addr, model.pc + 4094)
+
+        # Check that there are still any encodable address. (This will fail if
+        # the initial call had e.g. min_addr and max_addr were enormous and
+        # model.pc=0)
+        assert min_addr <= max_addr
+
         # We need to pick an out of bounds offset from all available values (A)
         # mode is a random variable to ensure that all the bins are hit
         # for offset range. It gives equal weight to all address ranges.
@@ -39,7 +50,9 @@ class BadBranch(BranchGen):
         elif mode <= 3 and program.imem_size - 1 <= max_addr:
             tgt_addr = max_addr
         elif mode <= 4 and program.imem_size - 1 <= max_addr:
-            tgt_addr = random.randrange(0, program.imem_size, 4) + 2
+            tgt_addr = random.randrange(max(0, min_addr),
+                                        min(program.imem_size, max_addr - 4),
+                                        4) + 2
         else:
             assert mode <= 4
             tgt_addr = random.randrange(min_addr + 2, max_addr + 1, 4)

